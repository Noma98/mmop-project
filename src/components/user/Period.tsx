import { CalendarIcon } from '@/components/icons';

type Props = {
  startDate: string;
  endDate: string;
};
export default function Period({ startDate, endDate }: Props) {
  return (
    <div className='flex gap-2 text-neutral-500 text-sm'>
      <div className='flex gap-2 items-center'>
        <CalendarIcon />
        <time>{startDate.slice(0, 7)}</time>
      </div>
      ~
      <div className='flex gap-2 items-center'>
        <CalendarIcon />
        <time>{endDate.slice(0, 7)}</time>
      </div>
    </div>
  );
}
