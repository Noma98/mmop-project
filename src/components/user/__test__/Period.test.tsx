import Period from '@/components/user/Period';
import { render, screen, waitFor } from '@testing-library/react';

describe('Period', () => {
  it('renders correctly', () => {
    const testStartDate = '2024-03-21';
    const testEndDate = '2024-05-09';
    const { asFragment } = render(
      <Period startDate={testStartDate} endDate={testEndDate} />
    );

    expect(screen.getByText('2024-03')).toBeInTheDocument();
    expect(screen.getByText('2024-05')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
