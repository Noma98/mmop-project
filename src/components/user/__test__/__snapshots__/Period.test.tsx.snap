// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Period renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex gap-2 text-neutral-500 text-sm"
  >
    <div
      class="flex gap-2 items-center"
    >
      <svg
        class="w-4 h-4"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M148 288h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm108-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 96v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm192 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96-260v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48zm-48 346V160H48v298c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
        />
      </svg>
      <time>
        2024-03
      </time>
    </div>
    ~
    <div
      class="flex gap-2 items-center"
    >
      <svg
        class="w-4 h-4"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M148 288h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm108-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 96v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm192 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96-260v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48zm-48 346V160H48v298c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
        />
      </svg>
      <time>
        2024-05
      </time>
    </div>
  </div>
</DocumentFragment>
`;
